runtime/bundles/org.eclipse.core.runtime/src/org/eclipse/core/internal/runtime/InternalPlatform.java

```diff
@@ -538,8 +540,8 @@ public IPath getStateLocation(Bundle bundle, boolean create) throws IllegalState
 	}
 
 	public long getStateTimeStamp() {
-		PlatformAdmin admin = getPlatformAdmin();
-		return admin == null ? -1 : admin.getState(false).getTimeStamp();
+		return Arrays.stream(getBundleContext().getBundles()).map(bundle -> bundle.adapt(Module.class))
+				.filter(Objects::nonNull).mapToLong(Module::getLastModified).sum();
```

Comment:
Just for the record: Theoretically it would be possible that the sum is unchanged if bundles are uninstalled but the timestamp increase of the remaining bundles exactly compensate that removal.
But such situation is very unlikely and as far as I can tell `Platform::getStateStamp` also makes no guaranties that it can never happen that the same number is returned twice.

Reply:
Just for the record, this method now has only one single consumer left that seem to handle somehow if/when some welcome contributions are scanned again :-D

Reply:
I'm not sure this is actually true? https://github.com/laeubi/eclipse.platform/blob/master/resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/ContentDescriptionManager.java#L548 but yeah, probabilities that this is going to happen, are rather low.

Reply:
I synced my fork to reflect current master and there you will see this

https://github.com/laeubi/eclipse.platform/blob/e7069db92859e15a44852ff753b9edeec2c1d7ed/resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/ContentDescriptionManager.java#L599

:-)